src/dune_rules/exe.mli

```diff
@@ -63,7 +65,9 @@ val link_many :
   -> dep_graphs Memo.t
 
 val build_and_link :
-     ?link_args:Command.Args.without_targets Command.Args.t Action_builder.t
+     ?link_args:
+       ([ `Linkall of bool ] * Command.Args.without_targets Command.Args.t)
```

Comment:
I don't understand the point of this argument. It seems like it's supposed to signify whether arguments conitain the `-linkall`, but then why not just check the list directly like in the old version of the PR?

Reply:
> but then why not just check the list directly like in the old version of the PR?

Because we don't have a list, we have a `Command.Args.without_targets Command.Args.t`.
The old version of this PR implemented a `command_args_has_flag` function which was incomplete and looked overkilled compared to the new implem.

Reply:
Indeed we don't have a list. 

What happens when a caller passes `(Linkall true, Command.Args.empty)`? If this is an error, do we have a check for this?

Reply:
We do not have a check for this. But I'm not sure how to check it without reintroducing the traversal of `Command.Args`.
 
Same issue with `(Linkall false, Command.Args.As "-kinall")`

Reply:
Yeah, I guess I'm still leaning towards checking `-linkall` inside the arg list.

In principle, I have no issue with the `Linkall` flag being passed, but it's weird that it has to be passed both in the argument list and as a flag. To me that's a poor API and a recipe for bugs later on.

A simple implementation that doesn't require your own flag traversal code would be to make `Command.expand` public and then use Action_builder directly to run the program.

Reply:
Now fixed. I was just a matter of exporting `Command.expand_no_targets`